squarescale/request.go

```diff
@@ -156,20 +156,23 @@ func (c *Client) request(method, path string, payload interface{}) (int, []byte,
 	return res.StatusCode, rbytes, err
 }
 
+type RequestError struct {
+	Error string `json:"error"`
+	DetailedError []string `json:"detailed_error"`
+}
+
 func unexpectedHTTPError(code int, body []byte) error {
 	// try to decode the body with { "error": "<description>" }
-	var description struct {
-		Error string `json:"error"`
-	}
+	var description RequestError
 
-	json.Unmarshal(body, description)
-	if description.Error != "" {
+	err := json.Unmarshal(body, &description)
+	if err == nil && description.Error != "" {
```

Comment:
You are sure you happy to continue if the Unmarshaller fails ? 

i.e. not returning errors if `err != nil` ?

Reply:
yes because with the current code, the Unmarshal fails (due to the missing & before `description` variable) and `description.Error` is empty therefore continues and still errors are handled/parsed with the following code when calling `decodeAnyError`